environment/src/main/java/com/sequenceiq/environment/environment/validation/securitygroup/gcp/GcpEnvironmentSecurityGroupValidator.java

```diff
@@ -70,18 +76,26 @@ private void validateSecurityGroup(EnvironmentDto environmentDto, ValidationResu
         CloudSecurityGroups securityGroups = platformParameterService.getSecurityGroups(request);
 
         boolean securityGroupFoundInRegion = false;
-        if (Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses())
-                && Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses().get(region.getName()))) {
-            for (CloudSecurityGroup cloudSecurityGroup : securityGroups.getCloudSecurityGroupsResponses().get(region.getName())) {
-                String groupId = cloudSecurityGroup.getGroupId();
-                if (groupId.equalsIgnoreCase(securityGroupId)) {
-                    securityGroupFoundInRegion = true;
-                    break;
+        Set<CloudSecurityGroup> cloudSecurityGroups = securityGroups.getCloudSecurityGroupsResponses().get(region.getName());
```

Comment:
This will yield an NPE if `securityGroups.getCloudSecurityGroupsResponses()` is `null`. I suggest wrapping this statement and `for` below inside a `if (Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses())` condition.